src/inputs/EditableLabel/EditableLabel.js

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { TextField, makeStyles, Typography, CircularProgress } from '@material-ui/core';
+import EditIcon from '@material-ui/icons/Edit';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    height: '100%',
+  },
+  labelDisplayContainer: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'stretch',
+  },
+  label: {
+    maxWidth: '500px',
+    color: theme.palette.primary.main,
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+  },
+  textField: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'stretch',
+    '& .MuiOutlinedInput-input': {
+      borderWidth: '2px',
+      outline: 'none',
+      padding: '8px',
+    },
+    '& .MuiOutlinedInput-root': {
+      minWidth: '300px',
+    },
+  },
+  editIcon: {
+    marginRight: '10px',
+    fontWeight: 'bold',
+    height: '20px',
+    color: theme.palette.primary.main,
+  },
+  circularLoading: {
+    marginLeft: '15px',
+    color: theme.palette.primary.contrastText,
+    backgroundColor: theme.palette.background.card,
+  },
+}));
+
+export const EditableLabel = (props) => {
+  const { value, onProcess, checkValue, labels } = props;
+
+  const classes = useStyles();
+
+  const [editMode, setEditMode] = useState(false);
+  const [error, setError] = useState(null);
+  const [isMouseHover, setMouseHover] = useState(false);
+  const [textFieldValue, setTextFiedlValue] = useState(value);
+
+  const startEdition = (event) => {
+    event.stopPropagation();
+    event.preventDefault();
+    setEditMode(true);
+  };
+
+  const modifyTextField = (event) => {
+    const newValue = event.target.value;
+    if (newValue !== value) {
+      setError(checkValue(newValue));
+    }
+  };
+
+  const processNewValue = (event) => {
+    event.stopPropagation();
+    const newValue = event.target.value;
+    if (newValue !== value) {
+      if (error == null) {
+        onProcess(newValue);
+        setTextFiedlValue(newValue);
+      }
+    }
+    setEditMode(false);
+    setMouseHover(false);
+  };
+
+  const labelDisplay = () => {
+    if (value === textFieldValue) {
+      return (
+        <div
+          className={classes.labelDisplayContainer}
+          onMouseLeave={() => setMouseHover(false)}
+          onMouseOver={() => setMouseHover(true)}
+          onClick={startEdition}
+        >
+          {isMouseHover && <EditIcon className={classes.editIcon} />}
+          <Typography className={classes.label} variant="h4">
```

Comment:
`variant="h4"` must not be hard-coded if we want this component to be generic. We should either add an optional prop (e.g. `typographyProps` or `labelProps`), or transfer unused props by using `...other` when destructuring props.